Constellation/Node/Api.hs

```diff
@@ -169,44 +228,51 @@ request allowSendReceive nvar req resp = do
                     (Shown $ Wai.remoteHost req, Shown apiReq)
                 resp unauthorized
 
-parseRequest :: [Text] -> BL.ByteString -> Either String ApiRequest
+parseRequest :: [Text] -> BL.ByteString -> RequestHeaders -> Either String ApiRequest
 -----
 -- Node client
 -----
-parseRequest ["send"]      b = ApiSend <$> AE.eitherDecode' b
-parseRequest ["receive"]   b = ApiReceive <$> AE.eitherDecode' b
+parseRequest ["send"]       b _ = ApiSend <$> AE.eitherDecode' b
+parseRequest ["receive"]    b _ = ApiReceive <$> AE.eitherDecode' b
+parseRequest ["sendRaw"]    b h = ApiSend <$> decodeSendRaw b h
+parseRequest ["receiveRaw"] b _ = ApiReceive <$> AE.eitherDecode' b
+parseRequest ["delete"]     b _ = ApiDelete <$> AE.eitherDecode' b
 -----
 -- Node to node
 -----
-parseRequest ["push"]      b = case decodeOrFail b of
+parseRequest ["push"]       b _ = case decodeOrFail b of
     Left  (_, _, err)  -> Left err
     Right (_, _, preq) -> Right $ ApiPush preq
-parseRequest ["resend"]    b = ApiResend <$> AE.eitherDecode' b
-parseRequest ["partyinfo"] b = case decodeOrFail b of
+parseRequest ["resend"]     b _ = ApiResend <$> AE.eitherDecode' b
+parseRequest ["partyinfo"]  b _ = case decodeOrFail b of
     Left  (_, _, err)   -> Left err
     Right (_, _, pireq) -> Right $ ApiPartyInfo pireq
 -----
 -- Miscellaneous
 -----
-parseRequest ["upcheck"]   _ = Right ApiUpcheck
-parseRequest _             _ = Left "Not found"
+parseRequest ["upcheck"]    _ _ = Right ApiUpcheck
+parseRequest _              _ _ = Left "Not found"
 
 authenticateRequest :: Bool -> ApiRequest -> Bool
 authenticateRequest False (ApiSend _)    = False
 authenticateRequest False (ApiReceive _) = False
 authenticateRequest _     _              = True
```

Comment:
Need to add sendRaw, receiveRaw and delete here as False (meaning they cannot be called via the external API)